### ISSUE TITLE
schools.stan example crashing R GUI on macOS Catalina

### ISSUE THREAD

**Issue by DanOvando** [stan-dev/rstan#789]
#### Summary:
Attempting to run schools example in the R GUI cashes the GUI

#### Description:

Running the standard schools.stan example from the rstan vignette. Running it in RStudio works (though now producing the message)

```
sh: clang++ -mmacosx-version-min=10.13: command not found
The NEXT version of Stan will not be able to pre-process your Stan program.
``` 

referenced in  https://github.com/stan-dev/rstan/issues/774

In trying to figure out what was up with that, I tried running the same example through the R GUI. After navigating to the correct directory and ensuring the code works in RStudio, running the below in R GUI results in the R GUI getting stuck indefinitely. The same thing happens if I try and run rstanarm with more than one core in the R GUI (GUI hangs indefinitely).

Actually, even if you force quite the R GUI, after this happens R keeps running in the background after the force quit, which I discovered from a very noisy fan.  

```r
schools <- "
data {
  int<lower=0> J;         // number of schools 
  real y[J];              // estimated treatment effects
  real<lower=0> sigma[J]; // standard error of effect estimates 
}
parameters {
  real mu;                // population treatment effect
  real<lower=0> tau;      // standard deviation in treatment effects
  vector[J] eta;          // unscaled deviation from mu by school
}
transformed parameters {
  vector[J] theta = mu + tau * eta;        // school treatment effects
}
model {
  target += normal_lpdf(eta | 0, 1);       // prior log-density
  target += normal_lpdf(y | theta, sigma); // log-likelihood
}

"

library(rstan)

schools_data <- list(
  J = 8,
  y = c(28,  8, -3,  7, -1,  1, 18, 12),
  sigma = c(15, 10, 16, 11,  9, 11, 10, 18)
)

library(rstan)
fit1 <- stan(
  model_code = schools,
  model_name = "schools",# Stan program
  data = schools_data,    # named list of data
  chains = 4,             # number of Markov chains
  warmup = 1000,          # number of warmup iterations per chain
  iter = 2000,            # total number of iterations per chain
  cores = 1,              # number of cores (could use one per chain)
  refresh = 100             
)
```

#### RStan Version:

2.21.1
#### R Version:
"R version 4.0.2 (2020-06-22)"

#### Operating System:
macOS 10.15.5

**Comment by jgabry** [comment]
@DanOvando sorry for the hassle and for the slow reply. I just noticed nobody responded to this. I think this is related to https://github.com/stan-dev/rstan/issues/774. What ended up working for people in that post was the following:

1. Remove the old RStan installation (`remove.packages("rstan")`)
2. Restart R 
3. Install RStan from GitHub with `remotes::install_github("stan-dev/rstan", ref = "develop", subdir = "rstan/rstan")`

Does that resolve the problem?

**Comment by DanOvando** [comment]
@jgabry no problem, it's not an issue that's holding me back but thought you'd like to know about it. Unfortunately that didn't do it. Was running the dev version of rstan already, and reinstalled following your steps to be sure, same problem persists when trying to run from the R GUI: it just freezes. Confirmed that same example runs fine in RStudio. rstan version 2.21.2

**Comment by DanOvando** [comment]
And looking at my activity monitor there are a bunch of rogue R processes running from the stuck R GUI runs that I killed

**Comment by jgabry** [comment]
Thanks for the letting us know. I'm glad it worked in RStudio but that's not good that we're crashing the R GUI. I'll leave this issue open for now then until there's a resolution for that.

**Comment by zsiders** [comment]
I'm experiencing the same issue. Mac OS Catalina 10.15.6 installed rStan following all the Catalina guidelines but when running the school example, the R GUI freezes and generates a lot of rogue R processes in the background. If I force quit all those processes occasionally a clang++ process starts consuming all the resources. Running R 4.0.2.

**Comment by jgabry** [comment]
Yeah this is super annoying but unfortunately I don't think we've figured out why this happens with the R GUI but not outside of the R GUI. Sorry! For now I would recommend using RStudio or running R from the terminal. Alternatively, if using the R GUI is important for you, then our new [CmdStanR](https://mc-stan.org/cmdstanr/) interface for R users might play nicer with the R GUI.
